File: Spark/Spark_with_scala/word_count/src/main/scala/word_count.scala
import org.apache.spark.{SparkConf, SparkContext}
import com.typesafe.config.{Config, ConfigFactory}
import org.apache.hadoop.fs._
import java.util.Calendar


object word_count {
  def main(args: Array[String]): Unit = {
    val props = ConfigFactory.load()
    val exMode = args(0)

    val conf = new SparkConf().
      setMaster(props.getConfig(exMode).getString("deployMode")).
      setAppName("Word Count")
    val sc = new SparkContext(conf)

    //Handling with file system
    val dataInPath = props.getConfig(exMode).getString("data.in.path")
    var dataOutPath = props.getConfig(exMode).getString("data.out.path")

    //HDFS API's
    val fs = FileSystem.get(sc.hadoopConfiguration)

    if(fs.exists(new Path(dataInPath)) == true){
      if(fs.exists(new Path(dataOutPath))){
        val hour = Calendar.getInstance().get(Calendar.HOUR_OF_DAY)
        val day = Calendar.getInstance().get(Calendar.DAY_OF_MONTH)
        val month = Calendar.getInstance().get(Calendar.MONTH)+1
        val year = Calendar.getInstance().get(Calendar.YEAR)
        dataOutPath = dataOutPath+"_"+year+"_"+month+"_"+day+"_"+hour
      }
      val textRDD = sc.textFile(dataInPath)
      val wordsRDD = textRDD.flatMap(line => line.split(" "))
      val wordCountRDD = wordsRDD.
        map(word => (word, 1)).
        reduceByKey( (v1: Int, v2: Int) => v1 + v2)
      wordCountRDD.saveAsTextFile(dataOutPath)
    }
    else {
      println("Input Path is not exist" + dataInPath)
    }
  }

}

File: Spark/Pyspark/word_count/src/main/python/word_count.py
import sys
import configparser as config
import datetime

from pyspark import SparkConf, SparkContext

props = config.RawConfigParser()
props.read("src/main/configs/app.properties")
exMode = sys.argv[1]

conf = SparkConf().\
    setMaster(props.get(exMode, 'deployMode')).\
    setAppName("Word Count")
sc = SparkContext(conf= conf)

#Handling with file system
dataInPath = props.get(exMode, 'data.in.path')
dataOutPath = props.get(exMode, 'data.out.path')

#HDFS API's
Path = sc._gateway.jvm.org.apache.hadoop.fs.Path
FileSytem = sc._gateway.jvm.org.apache.hadoop.fs.FileSystem
Configuration = sc._gateway.jvm.org.apache.hadoop.conf.Configuration

fs = FileSytem.get(Configuration())

if(fs.exists(Path(dataInPath)) == True):
    if(fs.exists(Path(dataOutPath))):
        dataOutPath = dataOutPath+"_"+str(datetime.datetime.now().strftime("%y_%m_%d_%H_%M"))
        print("New Path = ",dataOutPath)

    textRDD = sc.textFile(dataInPath)
    wordsRDD = textRDD.flatMap(lambda line: line.split(" "))
    wordCoutRDD = wordsRDD.map(lambda word: (word, 1)).reduceByKey(lambda v1, v2: v1 + v2)
    wordCoutRDD.saveAsTextFile(dataOutPath)
else:
    print("Input Path is not exist"+dataInPath)